src/ui/api/email.go

```diff
@@ -0,0 +1,117 @@
+/*
+   Copyright (c) 2016 VMware, Inc. All Rights Reserved.
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package api
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"strconv"
+
+	"github.com/vmware/harbor/src/common/api"
+	comcfg "github.com/vmware/harbor/src/common/config"
+	"github.com/vmware/harbor/src/common/dao"
+	"github.com/vmware/harbor/src/common/utils/email"
+	"github.com/vmware/harbor/src/common/utils/log"
+	"github.com/vmware/harbor/src/ui/config"
+)
+
+const (
+	pingEmailTimeout = 60
+)
+
+// EmailAPI ...
+type EmailAPI struct {
+	api.BaseAPI
+}
+
+// Prepare ...
+func (e *EmailAPI) Prepare() {
+	userID := e.ValidateUser()
+	isSysAdmin, err := dao.IsAdminRole(userID)
+	if err != nil {
+		log.Errorf("failed to check the role of user: %v", err)
+		e.CustomAbort(http.StatusInternalServerError, http.StatusText(http.StatusInternalServerError))
+	}
+
+	if !isSysAdmin {
+		e.CustomAbort(http.StatusForbidden, http.StatusText(http.StatusForbidden))
+	}
+}
+
+// Ping tests connection and authentication with email server
+func (e *EmailAPI) Ping() {
+	m := map[string]string{}
+	e.DecodeJSONReq(&m)
```

Comment:
So the request may look like: 
```
{
"host":"xxx", 
"port":xxx,
....}
```
Is this consistent with the ldap ping api?
I thought the request for ldap ping was like:
{"setting":{"host":"xxx" ....}}
could you double check?